utils/transitions/mixins.py

```diff
@@ -0,0 +1,114 @@
+from braces.views import MessageMixin
+from django.core.exceptions import ImproperlyConfigured
+from django.db import transaction
+from django.utils import six
+from django.utils.encoding import force_text
+from django.utils.functional import Promise
+
+
+class TransitionActionMixin:
+    transition_name = None
+    object_attribute = None
+
+    def get_transition_name(self):
+        return self.transition_name
+
+    def post(self, request, *args, **kwargs):
+        return self.perform_transition()
+
+    def perform_transition(self):
+        trans_object = self.get_transition_object()
+        transition = getattr(trans_object, self.get_transition_name())
+        # assert is transition
+        # TOOD: make this look more like form view mixinx
+        try:
+            with transaction.atomic():
+                transition(**self.get_transition_kwargs())
+        except Exception as exception:
+            self.handle_transition_exception(exception)
+            return self.transition_failure(exception)
+        # did we want to do different things based on the type of exception?
+        # except ToolException as e:
+        #     self.handle_transition_exception(e)
+        # except MachineError as e:
+        #     self.handle_transition_exception(e)
+        else:
+            return self.transition_success()
+
+    def handle_transition_exception(self, exception):
+        pass
+
+    def transition_success(self):
+        # raise NotImplementedError
+        pass
```

Comment:
I don't think i'm seeing where the transition_success/failure is actually transitioning a tool state.